### ISSUE TITLE
Large memory overhead of `ttnn.experimental.cumsum` with 1D tensors.

### ISSUE THREAD

**Issue by nmauriceTT** [tenstorrent/tt-metal#23264]
### Describe the bug

Currently, `ttnn.experimental.cumsum` handles accumulation on x and y axes by permuting axes. 

This approach can significantly increase the memory footprint of the tensor (by as much as a factor of 32).

### Steps to reproduce the issue

1. Create tensor of size [1, N] or [N, 1] 
2. Run `ttnn.experimental.cumsum(tensor, dim=0)` or `ttnn.experimental.cumsum(tensor, dim=1)` & measure memory footprint.

### Expected behavior

Memory footprint `ttnn.experimental.cumsum()` should not be more than that of its input and output.

**Comment by vzeljkovicTT** [comment]
## Additional case: 1D `[N]` tensor with `dim=0` — 1024x blowup

### Minimal repro

```python
import torch
import ttnn

N = 10_862_592  # size used in LLaVA-Next for cumsum(dim=0) on int32 positions

device = ttnn.open_device(device_id=0)
t = torch.zeros(N, dtype=torch.int32)
tt = ttnn.from_torch(
    t, dtype=ttnn.int32, layout=ttnn.TILE_LAYOUT,
    device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG
)
# Triggers: [N] → reshape [1,1,1,N] → permute [N,1,1,1]
# Peak allocation: N tiles = 10,862,592 × 1024 B = ~44 GB
out = ttnn.cumsum(tt, dim=0)
ttnn.close_device(device)
```

### Memory numbers for N = 10,862,592

| Stage | Bytes | Notes |
|---|---|---|
| Logical input | 43,450,368 B (43 MB) | N × 4 bytes |
| Input tiled `[1, N]` | 1,390,411,776 B (1.4 GB) | 1 × ⌈N/32⌉ tiles |
| Permuted `[N,1,1,1]` | **44,493,176,832 B (44 GB)** | N tiles — OOM on Wormhole |
